File: app/src/main/java/fi/hamk/riksu/myapplication/MyAdapter.java
package fi.hamk.riksu.myapplication;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import android.graphics.Typeface;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import org.joda.time.DateTime;

import java.util.List;


public class MyAdapter extends RecyclerView.Adapter<MyAdapter.MyHolder> {
    private Context c;
    //String s1[];// CUSTOM
    private Reservations reservations;
    String timeprefix;

    MyAdapter(Context ctx, Reservations reservations){       // s[]) {
        c = ctx;
        //s1=s;// CUSTOM
        this.reservations = reservations;
        timeprefix=c.getString(R.string.item_time_prefix);
    }

    @NonNull
    @Override
    public MyHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater mInf = LayoutInflater.from(c);
        View v = mInf.inflate(R.layout.my_row,parent,false);// CUSTOM my_row
        return new MyHolder(v);
    }
    //TODO: reservations
    @Override
    public void onBindViewHolder(@NonNull MyHolder holder, int position) {
        Reservation r = reservations.getReservations().get(position);
        List<ReservationResource> lista = r.getResources();

        String strGroups="";
        for (ReservationResource rr: lista ) {
            if(rr.getType().contains("student_group"))strGroups += (" "+rr.getName());
        }

        holder.tMore.setText(r.getSubject()+strGroups);// CUSTOMized
        StringBuilder s;
        s = new StringBuilder(RecViewHelper.formatDate(r.getStartDate()));
        s.append(" "+timeprefix+" ");   //s.append(" klo ");
        s.append(RecViewHelper.formatDate(r.getStartDate(),false));
        s.append(" - ");
        s.append(RecViewHelper.formatDate(r.getEndDate(),false));// End date defaults same as start date
        holder.t1.setText(s.toString() );// CUSTOMized

        // bold if today
        DateTime dt = new DateTime(r.getStartDate());
        DateTime now = DateTime.now();
        if(dt.toLocalDate().isEqual(now.toLocalDate()))
        {
            holder.t1.setTypeface(holder.t1.getTypeface(), Typeface.BOLD);
        }
    }

    @Override
    public int getItemCount() {
        return reservations.getReservations().size();
    }

    class MyHolder extends RecyclerView.ViewHolder {
        TextView t1;// CUSTOM
        TextView tMore;// CUSTOM
        MyHolder(View itemView) {
            super(itemView);
            t1 = itemView.findViewById(R.id.textView);// CUSTOM
            tMore = itemView.findViewById(R.id.textViewMore);// CUSTOM
        }
    }
}

File: app/src/main/java/fi/hamk/riksu/myapplication/MyAdapter.kt
package fi.hamk.riksu.myapplication

import android.content.Context
import androidx.recyclerview.widget.RecyclerView

import android.graphics.Typeface
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import kotlinx.android.extensions.LayoutContainer
import fi.hamk.riksu.myapplication.reservations.*
//import org.joda.time.DateTime

import kotlinx.android.synthetic.main.my_row.view.*
import java.time.LocalDateTime

class MyAdapter(private val c: Context, //String s1[];// CUSTOM
                 private val reservations: List<Any>? /*private val reservations: List<Reservation>?  */)
                 : RecyclerView.Adapter<MyAdapter.MyHolder>() {
    private val timeprefix = c.getString(R.string.item_time_prefix)

    init {       // s[]) {
    }//s1=s;// CUSTOM

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyHolder {
        val mInf = LayoutInflater.from(c)
        val v = mInf.inflate(R.layout.my_row, parent, false)// CUSTOM my_row
        return MyHolder(v, parent)
    }

    //TODO: reservations
    override fun onBindViewHolder(holder: MyHolder, position: Int) {
        reservations?.get(position)?.let { holder.bindReservations(it, timeprefix) }
    }

    override fun getItemCount(): Int {
        return reservations!!.size
    }

    class MyHolder(itemView: View, override val containerView: View?) : RecyclerView.ViewHolder(itemView), LayoutContainer {
        var t1: TextView = itemView.textView //findViewById(R.id.textView)// CUSTOM
        var tMore: TextView = itemView.textViewMore //findViewById(R.id.textViewMore)// CUSTOM

        fun bindReservations(r: Any/*Resource  Reservation */, timeprefix: String) {
            //val r:Reservation = reservations.reservations!![position]
            with(r) {
                if (r is Resource) {
                    val lista = this as Resource//resources

                    /*holder.*/ tMore.text = name //strGroups.insert(0, subject).toString()// CUSTOMized

                    /*holder.*/itemView.textView.text = code //s.toString()// was t1, CUSTOMized

                } else // Reservation
                {
                    val lista = this as Reservation

                    val strGroups = StringBuilder()
                    lista.resources?.forEach { rr -> // }
                        //for (rr in lista!!) {
                        if (rr.type!!.contains("student_group")) strGroups.append(" ").append(rr.name)
                    }

                    /*holder.*/ tMore.text = strGroups.insert(0, subject).toString()// CUSTOMized

                    val s: StringBuilder = StringBuilder(RecViewHelper.formatDate(startDate))
                    with(s) {
                        append(" ").append(timeprefix).append(" ")   //s.append(" klo ");
                        append(RecViewHelper.formatDate(startDate, false))
                        append(" - ")
                        append(RecViewHelper.formatDate(endDate, false))// End date defaults same as start date
                    }
                    /*holder.*/itemView.textView.text = s.toString()// was t1, CUSTOMized

                    // bold if today
                    val dt = LocalDateTime.parse(startDate) //DateTime(startDate)
                    val now = LocalDateTime.now()//DateTime.now()
                    if (dt.toLocalDate().isEqual(now.toLocalDate())) {
                        /*holder.*/t1.setTypeface(/*holder.*/t1.typeface, Typeface.BOLD)
                    }

                }
            }
        }
    }
}
